gloss/Graphics/Gloss/Geometry/Triangulation.hs

```diff
@@ -0,0 +1,388 @@
+
+
+-- | Map of events for processing. Holds segments that end in given $Vertex$ and segments that begin there.
+type Events = Map.Map Vertex (Set.Set Edge, Set.Set Edge)
+
+-- | Determen if points are close enough.
+(!=) :: Point -> Point -> Bool
+
+(!=) p1 p2 = Vertex p1 /= Vertex p2
+
+-- | Finds intersections of two Edges.
+intersection :: Maybe Edge -> Maybe Edge -> Maybe Vertex
+
+intersection (Just Edge{start = Vertex p0, end = Vertex p1}) (Just Edge{start = Vertex p2, end = Vertex p3})
+                | Just x <- intersectSegSeg p0 p1 p2 p3
+                , (x != p0) && (x != p1) && (x != p2) && ( x != p3)
+                = Just (Vertex x)
+                | otherwise = Nothing
+intersection _ _ = Nothing
+
+-- | Turns @Path@ into a graph in witch egdes can intersect. You can think of it as a drawing of a @Path@.
+makeInitialVertexSet :: Path -> Vertices
+
+makeInitialVertexSet [] = undefined
+makeInitialVertexSet points
+        = foldr accumulator Map.empty ( zip list (xs ++ list))
+           where accumulator (p , q) acc =
+                        let insert p1 q1 = Map.insertWith Set.union p1 (Set.singleton q1)
+                        in insert p q (insert q p acc)
+                 list@(_:xs) = map Vertex points
+
+
+-- | Calculates initial set of events.
+initialEvents :: Vertices -> Events
+
+initialEvents = Map.foldrWithKey accFn Map.empty
+                where accFn (Vertex p@(px,py)) neighbours events
+                        = let outCompare (Vertex (x,y)) = if px == x then py < y else px < x
+                              edgeStart point = Edge {start = Vertex p, end = point}
+                              edgeEnd point = Edge {end = Vertex p, start = point}
+                              startSet = Set.map edgeStart (Set.filter outCompare neighbours)
+                              endSet = Set.map edgeEnd (Set.filter (not.outCompare) neighbours)
+                          in Map.insert (Vertex p)  (startSet, endSet) events
+
+-- | Utility function for processing events. Similar to fold but the accumulator function also computes the new container.
+traverseVertices :: ((a, Map.Map b c) -> (b, c) -> (a, Map.Map b c)) -> a -> Map.Map b c -> a
+
+traverseVertices _ acc dictionary | Map.null dictionary = acc
+traverseVertices fn acc dictionary = let minElement = Map.findMin dictionary
+                                         ( newAcc, newDictionary) = fn  (acc, dictionary) minElement
+                                in traverseVertices fn  newAcc (Map.deleteMin newDictionary)
+
+-- | Accumulator type used for events processing.
+type AccumulatorType = (Set.Set Edge, Vertices)
+
+-- | Accumulator function used for events processing.
+accumulatorFunction :: (AccumulatorType , Events) -> (Vertex, (Set.Set Edge, Set.Set Edge)) -> (AccumulatorType, Events)
+
+accumulatorFunction ((edges, vertices), events) (_, (startOf, endOf)) = let deletedEdges = edges `Set.difference` (startOf  `Set.union` endOf)
```

Comment:
You'll need to reformat the code so it fits in 100 columns at most. Try to follow the coding style in rest of the Gloss library.